Engine/system/eventdispatcher.cpp

```diff
@@ -309,7 +309,7 @@ std::shared_ptr<Widget::Ref> EventDispatcher::implDispatch(Widget& root, FocusEv
   while(w->astate.focus!=nullptr) {
     w = w->astate.focus;
     }
-  if(w->wstate.focus) {
+  if(!event.in) {
```

Comment:
I don't think this is correct fix: `implDispatch` is called, only for foucs-out case, what means `!event.in` is always `true`.
Workaround I can propose, for X11 is to check, `focusLast` in `EventDispatcher::dispatchFocus` to be null, and if it's not - that mean we are waiting for FocusIn event

Reply:
So you basically mean to ignore the second FocusOut in `EventDispatcher::dispatchFocus`?

Reply:
Yes. There is no point of reporting `FocusOut` to game/ui twice .